llvm/lib/Transforms/IPO/ElimAvailExtern.cpp

```diff
@@ -45,6 +52,10 @@ void deleteFunction(Function &F) {
   ++NumRemovals;
 }
 
+static std::string getNewName(Module &M, const GlobalValue &GV) {
+  return GV.getName().str() + ".__uniq" + getUniqueModuleId(&M);
```

Comment:
nit: I meant also pull into the refactoring the setting of the name and the setting of the linkage type - basically reusing `convertToLocalCopy`

Reply:
I tried that route but not sure if that would be better. The new name is used at L85 as well. If we are gonna have a function that does both rename and set linkage, it also needs to return the new name for L85, which I find the semantics of it weird.